src/Jackett.Common/Indexers/CardigannIndexer.cs

```diff
@@ -880,7 +880,9 @@ public async Task<ConfigurationData> GetConfigurationForSetup(bool automaticlogi
                 configData.CookieHeader.Value = string.Join("; ", Login.Cookies);
             landingResult = await RequestWithCookiesAsync(LoginUrl.AbsoluteUri, referer: SiteLink);
 
-            if (landingResult.IsRedirect)
+            // Some sites have a temporary redirect before the login page, we need to process it.
+            // This check only apply to sites requiring it, which is currently: Yggtorrent
+            if (Definition.Id == "yggtorrent" && landingResult.IsRedirect)
```

Comment:
IMO a check that the 307 is occurring for just yggtorrent is the wrong approach.

any yml indexer that has the **followredirect: true**  statement 
https://github.com/Jackett/Jackett/blob/4df521562841e4142ee9fa0cb859db1fa71f1890/src/Jackett.Common/Definitions/yggtorrent.yml#L8
should allow for the 307 redirect to be accepted and processed.

this code just needs to confirm that the followredirect is enabled and then handle a 307 or reject it if not allowed.